### DISCUSSION TITLE
Add fallbacks for `gql.errorCode` and `gql.errorDetail` in graphql-dgs-spring-boot-micrometer

### DISCUSSION THREAD

**Discussion by caseygum** [Netflix/dgs-framework#1538] [Ideas]
Currently, the `gql.errorCode` and `gql.errorDetail` tags are extracted from `error.extensions.errorType` and `error.extensions.errorDetail` for generic GraphQLErrors in [DgsGraphQLMetricsInstrumentation](https://github.com/Netflix/dgs-framework/blob/4ad5f5eb74f0761abe521b6ab55cacee54f14273/graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/DgsGraphQLMetricsInstrumentation.kt#L378-L384).

While it's not strictly part of the GraphQL spec for error formatting, [the example](http://spec.graphql.org/June2018/#example-fce18) given on the spec, and the name of the tag itself would both suggest the key to look for in the `extensions` map would be `code`.  Apollo Server [also uses](https://www.apollographql.com/docs/apollo-server/data/errors/) `extensions.code`.

At Indeed, most of our almost 200 federated subgraphs use Netflix DGS, but have standardized on using `extensions.code` for error codes.  Some of our devs have complained that the tag values showing up in DataDog for `gql.errorCode` and `gql.errorDetail` are always `unknown` and `none`.  I would like to add a pull request to address where the DgsGraphQLMetricsInstrumentation.kt functions `errorTypeExtension()` and `errorDetailExtension()` are updated as follows:

```
   // look for extensions.errorType, then extensions.code, then error.errorType
    private fun <T : GraphQLError> errorTypeExtension(error: T): String {
        return extension(error, "errorType", extension(error, "code", error.errorType.toString()))
    }

    // look for extensions.errorDetail, then error.message
    private fun <T : GraphQLError> errorDetailExtension(error: T): String {
        return extension(error, "errorDetail", error.message ?: "none")
    }
```

Thoughts?

**Comment by srinivasankavitha** [comment]
Thanks for the detailed explanation. I think this makes sense.
